helmenv.sh

```diff
@@ -200,19 +217,31 @@ function helmenv_help() {
 }
 
 function helmenv_init () {
-    HELM_BINARY_PATH="$HOME/.bin"
-    ACTUAL_VERSION="$(basename "$(readlink -e "$HELM_BINARY_PATH/helm")")"
-    HELM_HOME="$HOME/.helm/${ACTUAL_VERSION//helm-}"
+    HELM_BINARY_PATH="${HELM_BINARY_PATH:-$HOME/.helm/bin}"
+    HELM_OS_ARCH="$(_helmenv_get_os_and_arch)"
+    if [[ "$HELM_OS_ARCH" == "darwin"* ]]; then
+        ACTUAL_VERSION="$(basename "$(greadlink -e "$HELM_BINARY_PATH/helm")")"
+    else
+        ACTUAL_VERSION="$(basename "$(readlink -e "$HELM_BINARY_PATH/helm")")"
+    fi
+    HELM_HOME="${HELM_HOME:-$HOME/.helm/${ACTUAL_VERSION//helm-}}"
 
     export HELM_HOME
+    export HELM_OS_ARCH
 
     if [[ ! -e "$HELM_HOME" ]]
     then
-        mkdir "$HELM_HOME"
+        mkdir -p "$HELM_HOME"
```

Comment:
Add `-p` to `mkdir` commands so intermediate directories can be created (in the event user-supplied `HELM_HOME` and `HELM_BINARY_PATH` directories do not exist).